### PR TITLE
Collect beta release notes

### PR DESCRIPTION
**Pull request by jochman**
fixes: https://github.com/demisto/etc/issues/20431
https://github.com/demisto/etc/issues/22491
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

The release-note creator will create `beta-release-notes.md` file in addition to the regular release notes.

### CODE REVIEW

File: release_notes.py [hunk 1]
```diff
@@ -43,7 +44,7 @@
 
 
 def add_dot(text):
-    if not text:
+    if not text or len(text) < 2:
```
Comment (yaakovi) [inline release_notes.py]: why?
dash release notes should not get to this point.

File: release_notes.py [hunk 2]
```diff
@@ -144,23 +146,25 @@ def load_data(self, data):
...
                     cnt = self.load_data(raw_content)
 
                     from_version = cnt.get("fromversion") or cnt.get("fromVersion")
                     to_version = cnt.get("toversion") or cnt.get("toVersion")
                     if from_version is not None and server_version_compare(current_server_version, from_version) < 0:
-                        print(f'Skipped because from version: {from_version}'
+                        print(f'{path}: Skipped because from version: {from_version}'
                               f' is greater than current server version: {current_server_version}')
+                        beta_rn_paths.append(path)
+                        print("added to rn paths")
```
Comment (yaakovi) [inline release_notes.py]: improve message or remove

File: release_notes.py [hunk 3]
```diff
@@ -179,9 +183,6 @@ def release_notes_section(self, store, title_prefix, current_server_version):
                     elif ans:
                         new_count += 1
                         new_str += ans
-                        print("Success")
```
Comment (yaakovi) [inline release_notes.py]: why did you remove this?

File: release_notes.py [hunk 4]
```diff
@@ -144,23 +146,25 @@ def load_data(self, data):
     # create a release notes section for store (add or modified) - return None if found missing release notes
     def release_notes_section(self, store, title_prefix, current_server_version):
         res = ""
+        beta_rn_paths = list()
         if store:
             new_str = ""
             new_count = 0
             for path in store:
                 with open(path, 'r') as file_obj:
-                    print(' - adding release notes ({}) for file - [{}]... '.format(path, title_prefix), end='')
```
Comment (yaakovi) [inline release_notes.py]: why did you remove this?

File: release_notes.py [hunk 5]
```diff
@@ -608,6 +612,13 @@ def create_content_descriptor(version, asset_id, res, github_token):
     with open('release-notes.md', 'w') as outfile:
         outfile.write(release_notes)
 
+    print("saving beta release notes")
+    with open('beta-release-notes.md', 'w') as outfile:
+        beta_release_notes = '## Demisto Content Beta Release Notes for version {} ({})\n'.format("5.5.0", asset_id)
```
Comment (yaakovi) [inline release_notes.py]:
```suggestion
        beta_release_notes = '## Demisto Content Beta Release Notes for version {} ({})\n'.format(version, asset_id)
```

```suggestion
        beta_release_notes = '## Demisto Content Beta Release Notes for version {}\n'.format(NEXT_VERSION)
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -175,7 +175,9 @@ jobs:
                 echo "Skipping - running only in container number 0"
                 exit 0
             fi
+            python copy_beta.py $CIRCLE_ARTIFACTS
             demisto-sdk create-content-artifacts -a $CIRCLE_ARTIFACTS
+            echo "Copying beta"
```
Comment (yaakovi) [inline .circleci/config.yml]: remove

File: .circleci/config.yml [hunk 7]
```diff
@@ -175,7 +175,9 @@ jobs:
                 echo "Skipping - running only in container number 0"
                 exit 0
             fi
+            python copy_beta.py $CIRCLE_ARTIFACTS
```
Comment (yaakovi) [inline .circleci/config.yml]: why is that a python script? just do it in bash
also, move this to release notes step

File: release_notes.py (hunk 1 above)
Comment (jochman) [reply release_notes.py]: It failed on some release notes on the next line somehow, maybe because of not enforcing changlog on some of 5.5 changes. it'll prevent the script from failing

File: release_notes.py (hunk 3 above)
Comment (jochman) [reply release_notes.py]: all the success messages removed because it made the log garbage out.

File: release_notes.py (hunk 4 above)
Comment (jochman) [reply release_notes.py]: because debugging and understanding the log became not possible

File: .circleci/config.yml (hunk 7 above)
Comment (jochman) [reply .circleci/config.yml]: its there until the `create_artifats` in demisto-sdk will be merged and released.
